### PR TITLE
Include latest Lambda Layer versions in documentation

### PR DESCRIPTION
**Pull request by nhinsch**
### What does this PR do?
Include the latest Lambda Layer versions directly in our documentation. Add a shortcode called `latest-lambda-layer-version` which prints out the version in order to make this maintainable going forward.

### Motivation
Currently, we frequently ask our users to click a link to GitHub or Amazon ECR Public to determine the latest version of a particular Lambda Layer. This is not ideal because:
- The user needs to leave our documentation and come back
- It can be hard to determine exactly what the latest layer version is. For example, the latest release of the dd-lambda-python is called `v3.56.0` in GitHub, but the layer version (what we care about) is just `56`.
It would be much simpler to just state the latest version number in our documentation. Because the version numbers appear in many places, we need to store the version number in a single place and then interpolate it into the documentation using a shortcode so that we can easily go back in and update the version numbers in the future.

### Preview
https://docs-staging.datadoghq.com/ngh/layer-versions/serverless/guide/datadog_forwarder_node
https://docs-staging.datadoghq.com/ngh/layer-versions/serverless/guide/datadog_forwarder_python
https://docs-staging.datadoghq.com/ngh/layer-versions/serverless/installation/dotnet
https://docs-staging.datadoghq.com/ngh/layer-versions/serverless/installation/java
https://docs-staging.datadoghq.com/ngh/layer-versions/serverless/installation/nodejs
https://docs-staging.datadoghq.com/ngh/layer-versions/serverless/installation/python
https://docs-staging.datadoghq.com/ngh/layer-versions/serverless/installation/ruby
https://docs-staging.datadoghq.com/ngh/layer-versions/serverless/libraries_integrations/extension

### Additional Notes
I explored and rejected an alternative solution using [shields.io](https://shields.io/). Shields.io is a service for creating "badges" with version numbers. The advantage of using shields.io is that the badges would update automatically to display the latest version numbers. However, I rejected the solution for two reasons:
- To use shields.io, we would need to expose an API endpoint in our production AWS account to tell shields.io what the latest layer versions are. This would likely take the form of a Lambda function and an API Gateway endpoint. I would rather not deploy and indefinitely maintain production infrastructure, even if it is relatively simple, in order to support this change.
- Shields.io creates badges, which are images. The images wouldn't fit into our documentation as seamlessly as text.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by agocs — APPROVED** [review]
So the thought is that we'll be manually updating `latest-lambda-layer-version.html` with every deploy going forward? Perhaps this (or at least a reminder) is something we could build into the release script.

**Comment by nhinsch** [comment]
Yeah, for now I will add a reminder, but going forward it would be good to find a way to automate this.

File: content/en/serverless/guide/datadog_forwarder_node.md [hunk 1]
```diff
@@ -84,17 +84,20 @@ Transform:
       env: "<ENV>" # Optional
 ```
 
-Replace `<SERVICE>` and `<ENV>` with appropriate values, `<LAYER_VERSION>` with the desired version of Datadog Lambda layer (see the [latest releases][4]), and `<FORWARDER_ARN>` with Forwarder ARN (see the [Forwarder documentation][2]).
+To fill in the placeholders:
 
-If your Lambda function is configured to use code signing, you must add Datadog's Signing Profile ARN (`arn:aws:signer:us-east-1:464622532012:/signing-profiles/DatadogLambdaSigningProfile/9vMI9ZAGLc`) to your function's [Code Signing Configuration][5] before you can use the macro.
+- Replace `<LAYER_VERSION>` with the desired version of the Datadog Lambda layer. The latest version is `{{< latest-lambda-layer-version layer="node" >}}`.
+- Replace `<FORWARDER_ARN>` with Forwarder ARN (see the [Forwarder documentation][2]).
+- Replace `<SERVICE>` and `<ENV>` with appropriate values.
```
Comment (apigirl) [inline content/en/serverless/guide/datadog_forwarder_node.md:91]:
```suggestion
- Replace `<SERVICE>` and `<ENV>` with your service and environment values.
```

File: content/en/serverless/guide/datadog_forwarder_node.md [hunk 2]
```diff
@@ -144,15 +146,18 @@ class CdkStack extends cdk.Stack {
 }
 ```
 
-Replace `<SERVICE>` and `<ENV>` with appropriate values, `<LAYER_VERSION>` with the desired version of Datadog Lambda layer (see the [latest releases][2]), and `<FORWARDER_ARN>` with Forwarder ARN (see the [Forwarder documentation][3]).
+To fill in the placeholders:
+
+- Replace `<LAYER_VERSION>` with the desired version of the Datadog Lambda layer. The latest version is `{{< latest-lambda-layer-version layer="node" >}}`.
+- Replace `<FORWARDER_ARN>` with Forwarder ARN (see the [Forwarder documentation][3]).
+- Replace `<SERVICE>` and `<ENV>` with appropriate values.
```
Comment (apigirl) [inline content/en/serverless/guide/datadog_forwarder_node.md:153]:
```suggestion
- Replace `<SERVICE>` and `<ENV>` with your service and environment values.
```

File: content/en/serverless/guide/datadog_forwarder_python.md [hunk 3]
```diff
@@ -79,7 +79,11 @@ Transform:
       env: "<ENV>" # Optional
 ```
 
-Replace `<SERVICE>` and `<ENV>` with appropriate values, `<LAYER_VERSION>` with the desired version of Datadog Lambda layer (see the [latest releases][4]), and `<FORWARDER_ARN>` with Forwarder ARN (see the [Forwarder documentation][2]).
+To fill in the placeholders:
+
+- Replace `<LAYER_VERSION>` with the desired version of the Datadog Lambda layer. The latest version is `{{< latest-lambda-layer-version layer="python" >}}`.
+- Replace `<FORWARDER_ARN>` with Forwarder ARN (see the [Forwarder documentation][2]).
+- Replace `<SERVICE>` and `<ENV>` with appropriate values.
```
Comment (apigirl) [inline content/en/serverless/guide/datadog_forwarder_python.md:86]:
```suggestion
- Replace `<SERVICE>` and `<ENV>` with your service and environment values.
```

File: content/en/serverless/guide/datadog_forwarder_python.md [hunk 4]
```diff
@@ -132,7 +136,11 @@ class CdkStack(core.Stack):
       })
 ```
 
-Replace `<SERVICE>` and `<ENV>` with appropriate values, `<LAYER_VERSION>` with the desired version of Datadog Lambda layer (see the [latest releases][4]), and `<FORWARDER_ARN>` with Forwarder ARN (see the [Forwarder documentation][2]).
+To fill in the placeholders:
+
+- Replace `<LAYER_VERSION>` with the desired version of the Datadog Lambda layer. The latest version is `{{< latest-lambda-layer-version layer="python" >}}`.
+- Replace `<FORWARDER_ARN>` with Forwarder ARN (see the [Forwarder documentation][2]).
+- Replace `<SERVICE>` and `<ENV>` with appropriate values.
```
Comment (apigirl) [inline content/en/serverless/guide/datadog_forwarder_python.md:143]:
```suggestion
- Replace `<SERVICE>` and `<ENV>` with your service and environment values.
```

File: content/en/serverless/guide/datadog_forwarder_node.md [hunk 5]
```diff
@@ -84,17 +84,20 @@ Transform:
       env: "<ENV>" # Optional
 ```
 
-Replace `<SERVICE>` and `<ENV>` with appropriate values, `<LAYER_VERSION>` with the desired version of Datadog Lambda layer (see the [latest releases][4]), and `<FORWARDER_ARN>` with Forwarder ARN (see the [Forwarder documentation][2]).
+To fill in the placeholders:
 
-If your Lambda function is configured to use code signing, you must add Datadog's Signing Profile ARN (`arn:aws:signer:us-east-1:464622532012:/signing-profiles/DatadogLambdaSigningProfile/9vMI9ZAGLc`) to your function's [Code Signing Configuration][5] before you can use the macro.
+- Replace `<LAYER_VERSION>` with the desired version of the Datadog Lambda layer. The latest version is `{{< latest-lambda-layer-version layer="node" >}}`.
```
Comment (apigirl) [inline content/en/serverless/guide/datadog_forwarder_node.md:89]: does using the version variable work in the docs?

Comment (apigirl) [reply content/en/serverless/guide/datadog_forwarder_node.md:89]: It totally does! I didn't realize we had implemented this as a partial. So cool.

Comment (apigirl) [reply content/en/serverless/guide/datadog_forwarder_node.md:89]: oooh you implement it in this PR. If I scrolled before commenting, I would be dangerous ;)

File: content/en/serverless/installation/python.md [hunk 6]
```diff
@@ -93,16 +93,16 @@ Transform:
...
       service: "<SERVICE>" # Optional
       env: "<ENV>" # Optional
 ```
 
 To fill in the placeholders:
 - Replace `<DATADOG_API_KEY>` with your Datadog API key from the [API Management page][4]. 
-- Replace `<LAYER_VERSION>` with the desired version of the Datadog Lambda layer (see the [latest releases][5]).
-- Replace `<EXTENSION_VERSION>` with the desired version of the Datadog Lambda Extension (see the [latest releases][6]).
+- Replace `<LAYER_VERSION>` with the desired version of the Datadog Lambda layer. The latest version is `{{< latest-lambda-layer-version layer="python" >}}`.
+- Replace `<EXTENSION_VERSION>` with the desired version of the Datadog Lambda Extension. The latest version is `{{< latest-lambda-layer-version layer="extension" >}}`.
```
Comment (Hesperide) [inline content/en/serverless/installation/python.md:105]: Could we just directly replace the values in the code snippet with the latest layer/extension versions?

File: content/en/serverless/installation/python.md [hunk 7]
```diff
@@ -263,7 +255,7 @@ For example:
             "DD_FLUSH_TO_LOG": "true",
             "DD_API_KEY": "<DATADOG_API_KEY>",
           },
-          "layers": ["arn:aws-us-gov:lambda:<AWS_REGION>:002406178527:layer:Datadog-Extension:<VERSION_NUMBER>"],
+          "layers": ["arn:aws-us-gov:lambda:<AWS_REGION>:002406178527:layer:Datadog-Extension:<EXTENSION_VERSION>"],
```
Comment (Hesperide) [inline content/en/serverless/installation/python.md:258]: Could we embed the latest version here to make things even easier?

File: layouts/shortcodes/latest-lambda-layer-version.html [hunk 8]
```diff
@@ -0,0 +1,30 @@
+<!--
```
Comment (Hesperide) [inline layouts/shortcodes/latest-lambda-layer-version.html:1]: This is super cool

**Review by davidejones — APPROVED** [review]
`latest-lambda-layer-version.html` looks good from websites point of view

File: content/en/serverless/installation/python.md (hunk 6 above)
Comment (nhinsch) [reply content/en/serverless/installation/python.md:105]: Yes, we could.

Advantage: no need to copy-paste the value.
Disadvantage: less obvious that you can choose a different value.

What do you think?

Comment (Hesperide) [reply content/en/serverless/installation/python.md:105]: Let's make it easy to copy paste, but maybe keep a note that the latest Extension version has been pre-populated and will need to be updated in the future

**Review by cswatt — APPROVED** [review]

**Comment by tianchu** [comment]
@nhinsch where did you add the reminder? could you add it to the wiki pages for our libraries release steps until it's automated?

**Comment by nhinsch** [comment]
@tianchu I have added instructions to the wiki pages for the layers.

**Pull request merged**